### PR TITLE
Adding Table of Content to Readme.md

### PR DESCRIPTION
**Pull request by ng29**
Added table to contents in the readme file

#### Summary

Table of contents has been added to readme.md for better user experience.
#### Ticket Link

For issue #3993

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,19 @@ This repository generates the documentation available at https://docs.mattermost
...
+
+ * [Contributing](#contributing)
+     * [Getting Started](#getting-started)
+     * [Editing](#editing)
+     * [Creating Pull Requests](#creating-pull-requests)
+     * [Using Labels](#using-labels)
+     * [Commenting on Pull Requests](#commenting-on-pull-requests)
+     * [Reviewing Pull Requests](#reviewing-pull-requests)
+ * [Building and Validating](#building-and-validating)
+
```
Comment (justinegeffen) [inline README.md:17]: The only suggestion I have is to delete the extraneous line breaks. :)

**Comment by justinegeffen** [comment]
This is really awesome, thank you @ng29! One minor suggestion but otherwise it's good to go. :)

File: README.md (hunk 1 above)
Comment (ng29) [reply README.md:17]: could you please specify that at which line you are referring?

Comment (justinegeffen) [reply README.md:17]: Lines 17 and 18 can be deleted (i.e., only one line-break between TOC and next header). :)

Comment (ng29) [reply README.md:17]: Modified

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Awesome, thanks again, @ng29! :)

**Comment by ng29** [comment]
> Awesome, thanks again, @ng29! :)

Glad to contribute, can't wait for the limited edition mattermost mug :)

**Pull request merged**